modules/hcp-ecs-client/services.tf

```diff
@@ -125,82 +88,52 @@ resource "aws_ecs_service" "frontend" {
   cluster         = aws_ecs_cluster.clients.arn
   task_definition = module.frontend.task_definition_arn
   desired_count   = 1
+
   network_configuration {
-    subnets          = var.private_subnet_ids
-    assign_public_ip = true
-    security_groups  = [var.security_group_id]
+    subnets         = var.private_subnet_ids
+    security_groups = [var.security_group_id]
   }
+
   load_balancer {
     target_group_arn = aws_lb_target_group.frontend.arn
     container_name   = "frontend"
     container_port   = local.frontend_port
   }
+
   launch_type            = "FARGATE"
   propagate_tags         = "TASK_DEFINITION"
   enable_execute_command = true
 }
 
-resource "aws_iam_role" "public_api-task-role" {
-  name = "public_api_${local.scope}_task_role"
-  assume_role_policy = jsonencode({
-    Version = "2012-10-17"
-    Statement = [
-      {
-        Action = "sts:AssumeRole"
-        Effect = "Allow"
-        Principal = {
-          Service = "ecs-tasks.amazonaws.com"
-        }
-      }
-    ]
-  })
-}
-
-resource "aws_iam_role" "public_api-execution-role" {
```

Comment:
as far as I can tell, these are the same as the roles made by default by the `mesh-task` resource: https://github.com/hashicorp/terraform-aws-consul-ecs/blob/v0.3.0/modules/mesh-task/iam.tf#L51

Reply:
yes, except they have the `local.scope` bit so that you don't run into conflicts. that should stay.

Reply:
ah ack and that makes sense. I saw `family` and thought that'd be enough, but can see how we'd want the random bit as well. Will addback